src/helpers/git_helpers/branch_configuration_helpers.sh

```diff
@@ -145,27 +145,34 @@ function ensure_knows_parent_branches {
           fi
         done
 
-        echo
-        echo -n "Please specify the parent branch of $(echo_inline_cyan_bold "$current_branch") by name or number (default: $MAIN_BRANCH_NAME): "
-        read parent
-        re='^[0-9]+$'
-        if [[ $parent =~ $re ]] ; then
-          # user entered a number here
-          parent=${branch_numbers[$parent]}
-          if [ -z "$parent" ]; then
-            echo_error_header
-            echo_error "Invalid branch number"
-            exit_with_error newline
+        local has_branch=false
+        while [ $has_branch == false ]; do
+          echo
+          echo -n "Please specify the parent branch of $(echo_inline_cyan_bold "$current_branch") by name or number (default: $MAIN_BRANCH_NAME): "
+          read parent
+          re='^[0-9]+$'
+          if [[ $parent =~ $re ]] ; then
+            # user entered a number here
+            parent=${branch_numbers[$parent]}
+            if [ -z "$parent" ]; then
+              echo_error_header
+              echo_error "Invalid branch number"
+            else
+              has_branch=true
+            fi
+          elif [ -z "$parent" ]; then
+            # user entered nothing
+            parent=$MAIN_BRANCH_NAME
+            has_branch=true
+          else
+            if [ "$(has_branch "$parent")" == "false" ]; then
+              echo_error_header
+              echo_error "branch '$parent' doesn't exist"
+            else
+              has_branch=true
+            fi
```

Comment:
I agree. I have thought about it for a couple of days now (and while I was writing this PR), and haven't found a good way to break this up into pieces that make sense by themselves. I have taken a note, and will get back to this later, once I have a better idea.